nexus_motion_planner/src/motion_plan_cache.cpp

```diff
@@ -0,0 +1,920 @@
+#include <memory>
+#include <vector>
+
+#include "moveit/robot_state/conversions.h"
+#include "moveit/robot_state/robot_state.h"
+#include "warehouse_ros/database_loader.h"
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/logging.hpp"
+#include "motion_plan_cache.hpp"
+
+namespace nexus {
+namespace motion_planner {
+
+#define NEXUS_MATCH_RANGE(arg, tolerance) \
+  arg - tolerance / 2, arg + tolerance / 2
+
+using warehouse_ros::MessageWithMetadata;
+using warehouse_ros::Metadata;
+using warehouse_ros::Query;
+
+MotionPlanCache::MotionPlanCache(const rclcpp::Node::SharedPtr& node)
+: node_(node)
+{
+  // TODO: methyldragon -
+  //   Declare or pass in these in the motion_planner_server instead?
+  if (!node_->has_parameter("warehouse_plugin"))
+  {
+    node_->declare_parameter<std::string>(
+      "warehouse_plugin", "warehouse_ros_sqlite::DatabaseConnection");
+  }
+
+  tf_buffer_ =
+    std::make_unique<tf2_ros::Buffer>(node_->get_clock());
+  tf_listener_ =
+    std::make_shared<tf2_ros::TransformListener>(*tf_buffer_);
+
+  warehouse_ros::DatabaseLoader loader(node_);
```

Comment:
I don't think we should use the `DatabaseLoader` class as it is forcing us to store (and spin) a separate ROS node in the MotionPlanner class. Looking at it's [implementation](https://github.com/ros-planning/warehouse_ros/blob/ros2/src/database_loader.cpp), it is a simple wrapper to retrieve ROS Params and load the right storage plugin. We're already retrieving some of the params in the `MotionPlanner` class, we can also try loading the plugin within the `on_configure()` and directly instantiate the `db` object via pluginlib. This would save us the need to spin another node which brings the total node count to three! (we already have one for this server, one for the move_group_interface).

Reply:
We still need a node for the tf listener..
I will just use the motion planner's node for both the tf listener and the database loading.